src/process_utility.c

```diff
@@ -1249,54 +1423,126 @@ process_index_end(Node *parsetree, CollectedCommand *cmd)
 	CatalogSecurityContext sec_ctx;
 	Cache	   *hcache;
 	Hypertable *ht;
-	bool		handled = false;
+	CreateIndexInfo info = {
+		.stmt = stmt,
+		.extended_options = hypertable_index_options,
+	};
+
+	hypertable_index_options = hypertable_index_options_default;
```

Comment:
@cevian the problem is that we don't have great way to pass `hypertable_index_options` form index_start to index_end, so right now I'm using a static variable. I want to clobber the static variable ASAP, so that options from one call to `CREATE INDEX` don't affect later calls. I _think_ I could add defaults to the `WithClauseArgs`; I'll keep an auxiliary array containing which args have not been called, and after I run out of arguments iterate through all of them and apply the default 🤔; but even so, I'll probably want to leave this line here for cases when the parser is not called: we don't want to call the parser in the non-hypertable case so postgres returns an "unknown argument" error when attempting to use `hypertable.foo` on a non-hypertable (I'll add a comment)